Tests/_URLRoutingTests/URLRoutingTests.swift

```diff
@@ -12,6 +12,21 @@ class URLRoutingTests: XCTestCase {
     XCTAssertEqual(try Method.post.print(), URLRequestData(method: "POST"))
   }
 
+  func testPath() {
+    XCTAssertEqual(123, try Path { Int.parser() }.parse(URLRequestData(path: "/123")))
+    XCTAssertThrowsError(try Path { Int.parser() }.parse(URLRequestData(path: "/123-foo"))) { error in
+      XCTAssertEqual(
+        """
+        error: unexpected input
+         --> input:1:5
+        1 | /123-foo
+          |     ^ expected end of input
```

Comment:
Before the `@_disfavoredOverload` fix, this succeeded by favoring the `parse` overload that doesn't check for the `End`.